core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -5302,24 +5310,30 @@ lower_select(JitCompContext *cc, x86::Assembler &a, COND_OP op, JitReg r0,
 }
 
 /* jmp to dst label */
-#define JMP_TO_LABEL(label_dst, label_src)                                   \
-    do {                                                                     \
-        if (label_is_ahead(cc, label_dst, label_src)) {                      \
-            char *stream = (char *)a.code()->sectionById(0)->buffer().data() \
-                           + a.code()->sectionById(0)->buffer().size();      \
-            int32 _offset = label_offsets[label_dst]                         \
-                            - a.code()->sectionById(0)->buffer().size();     \
-            Imm imm(INT32_MAX);                                              \
-            a.jmp(imm);                                                      \
-            /* The offset written by asmjit is always 0, we patch it again,  \
-               6 is the size of jmp instruciton */                           \
-            *(int32 *)(stream + 2) = _offset - 6;                            \
-        }                                                                    \
-        else {                                                               \
-            if (!jmp_from_label_to_label(a, jmp_info_list, label_dst,        \
-                                         label_src))                         \
-                GOTO_FAIL;                                                   \
-        }                                                                    \
+#define JMP_TO_LABEL(label_dst, label_src)                                 \
+    do {                                                                   \
+        if (label_is_ahead(cc, label_dst, label_src)) {                    \
+            JitErrorHandler *err_handler =                                 \
+                (JitErrorHandler *)a.code()->errorHandler();               \
+            int32 _offset;                                                 \
+            char *stream;                                                  \
+            Imm imm(INT32_MAX);                                            \
+            a.jmp(imm);                                                    \
+            if (!err_handler->err) {                                       \
```

Comment:
What kind of scenario would lead to the error? L5322 and L5332 seem to suggest both situations are acceptable.

Reply:
a.jmp(imm) may return error. The asmjit might calls re-alloc the memory of its code buffer, I tested cases with valgrind, and found that it sometimes reported that `*(int32 *)(stream + 2) = _offset` writes the old/obsolete memory before re-alloc, so I refactored the code. Another issue is that if a.jmp fails, we will write offset the invalid place.